yt_dlp/YoutubeDL.py

```diff
@@ -3500,7 +3509,7 @@ def correct_ext(filename, ext=new_ext):
                             info_dict['__real_download'] = info_dict['__real_download'] or real_download
                             success = success and partial_success
 
-                    if downloaded and merger.available and not self.params.get('allow_unplayable_formats'):
+                    if downloaded and should_merge:
```

Comment:
this is wrong: `should_merge` will always be truthy if `fd` is truthy; not the same logic as the old code